lib/HLSL/ComputeViewIdStateBuilder.cpp

```diff
@@ -925,3 +927,47 @@ ModulePass *createComputeViewIdStatePass() {
 }
 
 } // end of namespace llvm
+
+namespace {
+class ViewIdStatePrinter : public ModulePass {
```

Comment:
I'm not very familiar with these printer passes, but it looks to me that we have very few of them explicitly specified. Instead, if you add a `print()` method to the `ComputeViewIdState` pass and invoke `opt` with `-viewid-state -analyze`, I think the `opt.cpp` code will invoke your print function using the `ModulePassPrinter` class

Reply:
Never know that. New things learned.
Fixed.